src/rpc/blockchain.cpp

```diff
@@ -1916,6 +1923,435 @@ static UniValue savemempool(const JSONRPCRequest& request)
     return NullUniValue;
 }
 
+/** RAII object to prevent concurrency issue when scanning the txout set */
+static std::mutex g_utxosetscan;
+static std::atomic<int> g_scan_progress;
+static std::atomic<bool> g_scan_in_progress;
+static std::atomic<bool> g_should_abort_scan;
+class CoinsViewScanReserver
+{
+private:
+    bool m_could_reserve;
+public:
+    explicit CoinsViewScanReserver() : m_could_reserve(false) {}
+
+    bool reserve() {
+        assert (!m_could_reserve);
```

Comment:
`include <assert.h>`
nit: whitespace could benefit from clang-format

Reply:
Fixed

Reply:
Commit "Blockchain/RPC: Add scantxoutset method to scan UTXO set"

Missing fix :trollface: